### ISSUE TITLE
BOINC fails to connect to project/manager with only IPv6 resolver

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1017]
**Reported by gawul00 on 5 Feb 41072193 15:33 UTC**
BOINC is failing to use Mac's resolver libraries and fails to properly implement IPv6 resolver internally.

My Setup:
Mac OSX 10.6.6, BOINC 6.12.11 or 6.10.58, DNS resolvers: IPv6 2001:470:882a::1 and 2001:470:20::2

BOINC reports unable to connect to internet:

```
Tue Jan 25 22:55:28 2011     [HTTP_OP::init_get(): http://bam.boincstats.com/get_project_config.php
Tue Jan 25 22:55:33 2011     [http_debug](http_debug]) [Info: Expire cleared
Tue Jan 25 22:55:53 2011     [http_debug](ID#2]) [Info: Expire cleared
Tue Jan 25 22:55:53 2011     [http_debug](ID#2]) [Info: Could not resolve host: bam.boincstats.com (Timeout while contacting DNS servers)
Tue Jan 25 22:55:53 2011     [http_debug](ID#2]) [Info: Closing connection #0
Tue Jan 25 22:55:53 2011     [http_debug](ID#2]) HTTP error: Couldn't resolve host name
Tue Jan 25 22:55:54 2011     Project communication failed: attempting access to reference site
Tue Jan 25 22:55:54 2011     [HTTP_OP::init_get(): http://www.google.com/
Tue Jan 25 22:56:00 2011     [http_debug](http_debug]) [Info: Expire cleared
Tue Jan 25 22:56:20 2011     [http_debug](ID#0]) [Info: Expire cleared
Tue Jan 25 22:56:20 2011     [http_debug](ID#0]) [Info: Could not resolve host: www.google.com (Timeout while contacting DNS servers)
Tue Jan 25 22:56:20 2011     [http_debug](ID#0]) [Info: Closing connection #0
Tue Jan 25 22:56:20 2011     [http_debug](ID#0]) HTTP error: Couldn't resolve host name
Tue Jan 25 22:56:21 2011     BOINC can't access Internet - check network connection or proxy configuration.
```

Adding an IPv4 resolver 8.8.8.8, now connect works:

```
Tue Jan 25 22:59:39 2011     Fetching configuration file from http://bam.boincstats.com/get_project_config.php
Tue Jan 25 22:59:39 2011     [HTTP_OP::init_get(): http://bam.boincstats.com/get_project_config.php
Tue Jan 25 22:59:44 2011     [http_debug](http_debug]) [Info: Expire cleared
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Info: About to connect() to bam.boincstats.com port 80 (#0)
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Info: Trying 217.67.244.101... 
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Info: Connected to bam.boincstats.com (217.67.244.101) port 80 (#0)
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Sent header to server: GET /get_project_config.php HTTP/1.1
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Sent header to server: User-Agent: BOINC client (x86_64-apple-darwin 6.10.58)
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Sent header to server: Host: bam.boincstats.com
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Sent header to server: Accept: */*
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Sent header to server: Accept-Encoding: deflate, gzip
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Sent header to server: Content-Type: application/x-www-form-urlencoded
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Sent header to server: 
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: HTTP/1.1 200 OK
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: Date: Wed, 26 Jan 2011 04:59:45 GMT
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: Server: Apache
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: X-Powered-By: PHP/5.2.6-1+lenny9
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: Content-Encoding: gzip
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: Vary: Accept-Encoding
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: Content-Length: 189
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: Content-Type: application/xml; charset=ISO-8859-1
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Received header from server: 
Tue Jan 25 22:59:45 2011     [http_xfer_debug](ID#2]) [HTTP: wrote 227 bytes
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Info: Expire cleared
Tue Jan 25 22:59:45 2011     [http_debug](ID#2]) [Info: Connection #0 to host bam.boincstats.com left intact
Tue Jan 25 23:01:00 2011     [http_debug](ID#2]) HTTP_OP::init_post(): http://bam.boincstats.com/rpc.php
Tue Jan 25 23:01:00 2011     Contacting account manager at http://bam.boincstats.com/
Tue Jan 25 23:01:00 2011     [[ID#0](http_debug]) Info: Connection #0 seems to be dead!
Tue Jan 25 23:01:00 2011     [[ID#0](http_debug]) Info: Expire cleared
Tue Jan 25 23:01:00 2011     [[ID#0](http_debug]) Info: Closing connection #0
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Info: Expire cleared
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Info: About to connect() to bam.boincstats.com port 80 (#0)
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Info: Trying 217.67.244.101... 
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Info: Connected to bam.boincstats.com (217.67.244.101) port 80 (#0)
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: POST /rpc.php HTTP/1.1
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: User-Agent: BOINC client (x86_64-apple-darwin 6.10.58)
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: Host: bam.boincstats.com
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: Accept: */*
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: Accept-Encoding: deflate, gzip
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: Content-Type: application/x-www-form-urlencoded
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: Content-Length: 305
Tue Jan 25 23:01:05 2011     [[ID#0](http_debug]) Sent header to server: 
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: HTTP/1.1 200 OK
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: Date: Wed, 26 Jan 2011 05:01:05 GMT
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: Server: Apache
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: X-Powered-By: PHP/5.2.6-1+lenny9
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: Content-Encoding: gzip
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: Vary: Accept-Encoding
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: Content-Length: 483
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: Content-Type: application/xml; charset=UTF-8
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Received header from server: 
Tue Jan 25 23:01:06 2011     [[ID#0](http_xfer_debug]) HTTP: wrote 973 bytes
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Info: Expire cleared
Tue Jan 25 23:01:06 2011     [[ID#0](http_debug]) Info: Connection #0 to host bam.boincstats.com left intact
Tue Jan 25 23:01:06 2011     Account manager: BAM! User-ID: 17439
Tue Jan 25 23:01:06 2011     Account manager: BAM! Host-ID: 261537
Tue Jan 25 23:01:06 2011     Account manager: Number of BAM! connections for this host: 1
Tue Jan 25 23:01:06 2011     Account manager contact succeeded
```

**Comment by romw** [comment]
**Commented by charlief on 26 Feb 41096472 16:53 UTC**
I am not able to reproduce this, but I'm not sure if I'm setting my network configurations the same as yours.  I am setting "Configure IPv" to "Using DHCP" (since I am behind a router with NAT), setting "DNS server; to "2001:470:882a::1, 2001:470:20::2" and leaving "Search domains" blank.  If this is not correct, please post the full procedure for setting the Mac to reproduce this.

Also, please explain why you would want to use an "only IPv6 resolver", given the many Internet servers that do not support IPv6.

Thank you.

**Comment by romw** [comment]
**Commented by gawul00 on 8 Jan 41106504 04:53 UTC**
Replying to [charlief](comment:2):

I'll start with the second question first.  The IPv6 resolver is a caching nameserver that is running in my network.  It receives requests from my local machines and if the requested address is not available in the cache, it contacts the nameserver hierarchy to resolve the name to an IP address.  The address requested may be an IPv4 or an IPv6 address depending on the  application making the request.  The only portion that is using IPv6 exclusively is transport between my local desktop and the caching nameserver on my local network.  The caching nameserver has both IPv4 and IPv6 connectivity to the Internets for resolving names.  The local machine also has IPv4 connectivity for communications to webservers and such.  I had been testing with using an IPv6 only resolver in my local IPv6 enabled network.

The two IPv6 addresses listed are:
2001:470:882a:1 - local IPv6 caching nameserver on my network (not publicly available)
2001:470:20::2 - publicly accessible IPv6 caching nameserver run by Hurricane Electric

Now on to the test setup.  

1. For you to replicate this bug, you will need to have IPv6 connectivity between your test Mac and either a local caching nameserver accessible over IPv6, or the IPv6 internet where you can access the Hurricane Electricn ameserver at 2001:470:20::2.  Confirm proper IPv6 connectivity for your Mac - it should show an IPv6 address just below the Search Domains box.

2. Then the DNS Server setting should be set to "2001:470:20::2" or to your local caching IPv6 nameserver, but only IPv6 addresses should be in the box.  

* You can confirm connectivity to the HE server by opening a terminal window and using "ping6 2001:470:20::2"

* You can confirm your machine is able to resolve names to IPs using this server by "dig A boinc.berkeley.edu @2001:470:20::2" which should return a CNAME and an A result in the ANSWER SECTION.  (An A record is a mapping between a name and IPv4 address. If an IPv6 address is needed, that is stored in a AAAA record.  CNAME is a name that maps to another name)

* As a final DNS check, you can use "nslookup boinc.berkeley.edu" to show what the operating system will do when asked for the IPv4 address related to the name boinc.berkeley.edu - it should show the nameserver it contacted, the address it used and the answer (nslookup will look up an A record by default, in this case a canonical name is returned and that name is resolved to an IPv4 address).

Here is the result from my machine:

```
oberon:~ gawul00$ nslookup boinc.berkeley.edu
Server:		2001:470:882a:200::2  <----------------------------- Request being made to this server
Address:	2001:470:882a:200::2#53   <------------------------- Server address and port

Non-authoritative answer:   <--------------------------------------- is an answer from my local resolver, not the Berkeley server directly
boinc.berkeley.edu	canonical name = boinc.SSL.berkeley.edu. <-- Request for boinc.berkeley.edu returned another c-name
Name:	boinc.SSL.berkeley.edu  <----------------------------------- Lookup the c-name record
Address: 128.32.18.189  <------------------------------------------- which returns this IP address

oberon:~ gawul00$ 
```

Now try to use Boinc - debugs show that it is unable to resolve names, however the OS continues to be able to do so:

```

oberon:~ gawul00$ nslookup bam.boincstats.com
Server:		2001:470:882a:200::2
Address:	2001:470:882a:200::2#53

Non-authoritative answer:
Name:	bam.boincstats.com
Address: 217.67.244.101

oberon:~ gawul00$ nslookup www.google.com
Server:		2001:470:882a:200::2
Address:	2001:470:882a:200::2#53

Non-authoritative answer:
www.google.com	canonical name = www.l.google.com.
Name:	www.l.google.com
Address: 74.125.93.105
Name:	www.l.google.com
Address: 74.125.93.147
Name:	www.l.google.com
Address: 74.125.93.106
Name:	www.l.google.com
Address: 74.125.93.99
Name:	www.l.google.com
Address: 74.125.93.104
Name:	www.l.google.com
Address: 74.125.93.103

oberon:~ gawul00$ 
```

**Comment by romw** [comment]
**Commented by charlief on 20 Mar 41110118 01:46 UTC**
Thank you for your replies.  Unfortunately, I have not been able to get ping6 to work on my system; I suspect it is a problem with my Internet modem / router (2-wire 3800HGV-B with AT&T U-Verse service), even though I've tried using DMZ-plus with a hard-wired Ethernet connection, and even though the router manufacturer claims it is IPv6-capable.  Ping6 reports ''UDP connect: No route to host''.

Rather than spending more time trying to reproduce this here, can I ask you to try the following version of BOINC? [http://boinc.berkeley.edu/dl/boinc_6.10.43_macOSX_universal.zip].

BOINC uses the libraries libcurl and c-ares to do its Internet communications.  c-ares 1.7.0 introduced asynchronous name resolving supporting IPv6, but caused problems on some network configurations so we reverted to c-ares 1.6.0 which did not have IPv6 support.  BOINC version 6.10.43 still used c-ares 1.7.0, so it would be great to know if that works with your IPv6 resolver.

If that works, I would then like to build you a test drop of BOINC built with a patched version of c-ares 1.7.0.  I have confirmed that my patch fixes the original issue with c-ares 1.7.0, but I have not been able to confirm that it does work properly with IPv6.

By coincidence (or perhaps not), the network configurations that failed with standard c-ares 1.7.0 were the same 2-wire 3800HGV-B router with AT&T U-Verse service. The problem occurs when the Mac's Network System Preferences are set to the default Airport settings, which use the router as the DNS Server.  If I use that configuration, I get the same failures.  The problem is that the standard c-ares 1.7.0 code doesn't fall back to trying IPv4 for the particular error it gets from the router's DNS server.  If I instead configure the Mac to use different (external) DNS servers, standard c-ares 1.7.0 works fine.  My patch (which was '''not''' in BOINC 6.10.43) simply makes c-ares fall back to IPv4 for any IPv6 failure.

Thanks in advance.

**Comment by romw** [comment]
**Commented by gawul00 on 23 Nov 41161100 17:46 UTC**
I had a chance to try out 6.10.43 and found the same behavior.  With only an IPv6 resolver in my network connection properties, BOINC fails to resolve DNS.  When I add an IPv4 resolver, I am able to connect successfully.


WIth only an IPv6 resolver configured:
```
Sun Feb 27 19:12:40 2011		Starting BOINC client version 6.10.43 for x86_64-apple-darwin
Sun Feb 27 19:12:40 2011		Libraries: libcurl/7.19.7 OpenSSL/0.9.7l zlib/1.2.3 c-ares/1.7.0
Sun Feb 27 19:12:40 2011		OS: Darwin: 10.6.0
Sun Feb 27 19:13:23 2011		Fetching configuration file from http://bam.boincstats.com/get_project_config.php
Sun Feb 27 19:13:23 2011		[HTTP_OP::init_get(): http://bam.boincstats.com/get_project_config.php
Sun Feb 27 19:13:23 2011		[proxy_debug](http_debug]) HTTP_OP::no_proxy_for_url(): http://bam.boincstats.com/get_project_config.php
Sun Feb 27 19:13:23 2011		[returning false
Sun Feb 27 19:13:27 2011		[http_debug](proxy_debug]) [Info:  Expire cleared
Sun Feb 27 19:13:44 2011		[http_debug](ID#2]) [Info:  Expire cleared
Sun Feb 27 19:13:48 2011		[http_debug](ID#2]) [Info:  Expire cleared
Sun Feb 27 19:13:59 2011		[http_debug](ID#2]) [Info:  Expire cleared
Sun Feb 27 19:13:59 2011		[http_debug](ID#2]) [Info:  Could not resolve host: bam.boincstats.com (Timeout while contacting DNS servers)
Sun Feb 27 19:13:59 2011		[http_debug](ID#2]) [Info:  Closing connection #0
Sun Feb 27 19:13:59 2011		[network_status_debug](ID#2]) got HTTP error - checking ref site
Sun Feb 27 19:13:59 2011		[HTTP error: Couldn't resolve host name
Sun Feb 27 19:14:00 2011		[network_status_debug](http_debug]) need_phys_conn 0; trying http://www.google.com/
Sun Feb 27 19:14:00 2011		Project communication failed: attempting access to reference site
Sun Feb 27 19:14:00 2011		[HTTP_OP::init_get(): http://www.google.com/
Sun Feb 27 19:14:00 2011		[proxy_debug](http_debug]) HTTP_OP::no_proxy_for_url(): http://www.google.com/
Sun Feb 27 19:14:00 2011		[returning false
Sun Feb 27 19:14:05 2011		[http_debug](proxy_debug]) [Info:  Expire cleared
Sun Feb 27 19:14:06 2011		[network_status_debug](ID#0]) status: reference site lookup pending
Sun Feb 27 19:14:17 2011		[[ID#0](http_debug]) Info:  Expire cleared
Sun Feb 27 19:14:18 2011		[status: reference site lookup pending
Sun Feb 27 19:14:21 2011		[http_debug](network_status_debug]) [Info:  Expire cleared
Sun Feb 27 19:14:22 2011		[network_status_debug](ID#0]) status: reference site lookup pending
Sun Feb 27 19:14:34 2011		[[ID#0](http_debug]) Info:  Expire cleared
Sun Feb 27 19:14:34 2011		[[ID#0](http_debug]) Info:  Could not resolve host: www.google.com (Timeout while contacting DNS servers)
Sun Feb 27 19:14:34 2011		[[ID#0](http_debug]) Info:  Closing connection #0
Sun Feb 27 19:14:34 2011		[HTTP error: Couldn't resolve host name
Sun Feb 27 19:14:34 2011		[network_status_debug](http_debug]) status: reference site lookup pending
Sun Feb 27 19:14:35 2011		BOINC can't access Internet - check network connection or proxy configuration.
```


With IPv4 resolver added:
```

Sun Feb 27 19:22:26 2011		Fetching configuration file from http://bam.boincstats.com/get_project_config.php
Sun Feb 27 19:22:26 2011		[HTTP_OP::init_get(): http://bam.boincstats.com/get_project_config.php
Sun Feb 27 19:22:26 2011		[proxy_debug](http_debug]) HTTP_OP::no_proxy_for_url(): http://bam.boincstats.com/get_project_config.php
Sun Feb 27 19:22:26 2011		[returning false
Sun Feb 27 19:22:26 2011		[http_debug](proxy_debug]) [Info:  About to connect() to bam.boincstats.com port 80 (#0)
Sun Feb 27 19:22:26 2011		[http_debug](ID#2]) [Info:    Trying 217.67.244.101... 
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Info:  Connected to bam.boincstats.com (217.67.244.101) port 80 (#0)
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Sent header to server: GET /get_project_config.php HTTP/1.1
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Sent header to server: User-Agent: BOINC client (x86_64-apple-darwin 6.10.43)
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Sent header to server: Host: bam.boincstats.com
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Sent header to server: Accept: */*
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Sent header to server: Accept-Encoding: deflate, gzip
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Sent header to server: Content-Type: application/x-www-form-urlencoded
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Sent header to server: 
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: HTTP/1.1 200 OK
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: Date: Mon, 28 Feb 2011 01:22:33 GMT
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: Server: Apache
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: X-Powered-By: PHP/5.2.6-1+lenny9
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: Content-Encoding: gzip
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: Vary: Accept-Encoding
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: Content-Length: 189
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: Content-Type: application/xml; charset=ISO-8859-1
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Received header from server: 
Sun Feb 27 19:22:27 2011		[http_xfer_debug](ID#2]) [HTTP: wrote 227 bytes
Sun Feb 27 19:22:27 2011		[http_debug](ID#2]) [Info:  Connection #0 to host bam.boincstats.com left intact
Sun Feb 27 19:22:30 2011		[http_debug](ID#2]) HTTP_OP::init_post(): http://bam.boincstats.com/rpc.php
Sun Feb 27 19:22:30 2011		[HTTP_OP::no_proxy_for_url(): http://bam.boincstats.com/rpc.php
Sun Feb 27 19:22:30 2011		[proxy_debug](proxy_debug]) returning false
Sun Feb 27 19:22:30 2011		Contacting account manager at http://bam.boincstats.com/
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Info:  Connection #0 seems to be dead!
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Info:  Expire cleared
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Info:  Closing connection #0
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Info:  About to connect() to bam.boincstats.com port 80 (#0)
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Info:    Trying 217.67.244.101... 
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Info:  Connected to bam.boincstats.com (217.67.244.101) port 80 (#0)
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: POST /rpc.php HTTP/1.1
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: User-Agent: BOINC client (x86_64-apple-darwin 6.10.43)
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: Host: bam.boincstats.com
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: Accept: */*
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: Accept-Encoding: deflate, gzip
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: Content-Type: application/x-www-form-urlencoded
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: Content-Length: 2827
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: Expect: 100-continue
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Sent header to server: 
Sun Feb 27 19:22:30 2011		[[ID#0](http_debug]) Received header from server: HTTP/1.1 100 Continue
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: HTTP/1.1 200 OK
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: Date: Mon, 28 Feb 2011 01:22:37 GMT
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: Server: Apache
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: X-Powered-By: PHP/5.2.6-1+lenny9
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: Content-Encoding: gzip
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: Vary: Accept-Encoding
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: Content-Length: 874
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: Content-Type: application/xml; charset=UTF-8
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Received header from server: 
Sun Feb 27 19:22:31 2011		[[ID#0](http_xfer_debug]) HTTP: wrote 2114 bytes
Sun Feb 27 19:22:31 2011		[[ID#0](http_debug]) Info:  Connection #0 to host bam.boincstats.com left intact
Sun Feb 27 19:22:31 2011		[unparsed_xml] ACCT_MGR_OP::parse: unrecognized version
Sun Feb 27 19:22:31 2011		Account manager: BAM! User-ID: 17439
Sun Feb 27 19:22:31 2011		Account manager: BAM! Host-ID: 261537
Sun Feb 27 19:22:31 2011		Account manager: Number of BAM! connections for this host: 42
Sun Feb 27 19:22:31 2011		Account manager contact succeeded
Sun Feb 27 19:22:31 2011		General prefs: from http://bam.boincstats.com/ (last modified 27-Feb-2011 19:22:31)
Sun Feb 27 19:22:31 2011		Host location: none
Sun Feb 27 19:22:31 2011		General prefs: using your defaults
Sun Feb 27 19:22:31 2011		Preferences:
Sun Feb 27 19:22:31 2011		   max memory usage when active: 8192.00MB
Sun Feb 27 19:22:31 2011		   max memory usage when idle: 8192.00MB
Sun Feb 27 19:22:31 2011		   max disk usage: 100.00GB
Sun Feb 27 19:22:31 2011		   max CPUs used: 4
Sun Feb 27 19:22:31 2011		   (to change, visit the web site of an attached project,
Sun Feb 27 19:22:31 2011		   or click on Preferences)
```
